Snowflake.Data/Core/HttpUtil.cs

```diff
@@ -238,17 +242,43 @@ void IRule.apply(NameValueCollection queryParams)
                 }
             }
 
+            /// <summary>
+            /// RetryReasonRule would update the retryReason parameter
+            /// </summary>
+            class RetryReasonRule : IRule
+            {
+                int retryReason;

```

Comment:
That's because in .NET the http error from response is integer (actually enum). https://github.com/snowflakedb/snowflake-connector-net/blob/SNOW-736347-retry-reason/Snowflake.Data/Core/HttpUtil.cs#L407
It will be convert to string when put into URL:
https://github.com/snowflakedb/snowflake-connector-net/blob/SNOW-736347-retry-reason/Snowflake.Data/Core/HttpUtil.cs#L264